pkgs/development/libraries/flutter-engine/default.nix

```diff
@@ -0,0 +1,399 @@
+                      dir=$(dirname $file | sed "s|$input/$path||g")
+                      mkdir -p $out/usr/$path/$dir
+                      cp -r --no-preserve=ownership,mode $file $out/usr/$path || true
+                    elif [[ ! -e $out/usr/$path/$(basename $file) ]]; then
+                      if [[ $path == */pkgconfig ]] && [[ -z $(grep "prefix=" $file) ]]; then
+                        echo "prefix=$input" | cat - $file > $out/usr/$path/$(basename $file)
+                      else
+                        ln -s $file $out/usr/$path
+                      fi
+                    fi
+                  done
+                fi
+              done
+            done <$closure/store-paths
+
+            # Wrap pkg-config
+            rm $out/usr/bin/pkg-config
+            cp ${pkg-config}/bin/pkg-config $out/usr/bin/pkg-config
+            wrapProgram $out/usr/bin/pkg-config \
+              --set PKG_CONFIG_PATH $out/usr/lib/pkgconfig \
+              --set PKG_CONFIG_FOR_TARGET $out/usr/bin/pkg-config \
+              --set PKG_CONFIG_FOR_TARGET $out/usr/lib/pkgconfig
+          '';
+        };
+
+        src = stdenvNoCC.mkDerivation {
+          pname = "flutter-engine-src";
+          version = builtins.substring 0 7 args.version;
+
+          passthru = {
+            inherit flutter-deps;
+          };
+
+          nativeBuildInputs = [ patchelf python3 makeWrapper ];
+
+          src = args.src or fetchFromGitHub {
+            owner = "flutter";
+            repo = "engine";
+            rev = args.version;
+            inherit sha256;
+          };
+
+          buildroot = fetchFromGitHub {
+            owner = "flutter";
+            repo = "buildroot";
+            rev = "6af51ff4b86270cc61517bff3fff5c3bb11492e1";
+            sha256 = "sha256-03jr3H1RvzGfa0rBPZ1rtNpmieKzDjDgBsrZGaj7vuI=";
+          };
+
+          dontUnpack = true;
+          dontPatch = true;
+          dontConfigure = true;
+          dontBuild = true;
+
+          ## FIXME: use "flutter-deps" and the arguments which the OG script takes to generate the git revision
+          gitrev = writeText "git-revision.py" ''
+            #!${python3}/bin/python3
+            print("${fakeHash}")
+          '';
+
+          dartPackageConfig = writeText "dart-package-config.json" ''
+            {
+              "configVersion": 2,
+              "generator": "tools/generate_package_config.dart",
+              "packages": []
+            }
+          '';
+
+          binaryFixes = [
+            "src/third_party/dart/tools/sdks/dart-sdk/bin/dart"
+            "src/flutter/third_party/gn/gn"
+          ];
+
+          shellFixes = [
+            "src/third_party/dart/tools/sdks/dart-sdk/bin/dart2js"
+            "src/flutter/tools/gn"
+          ];
+
+          patches = [
+            ./gn.patch # Disables angle tests via gn
+            ./atk-object.patch # Fixes redeclaration issue (https://github.com/flutter/engine/pull/36056)
```

Comment:
the PR was merged 3 months ago, lets just update src

Reply:
Should make sure that's a part of a stable release of the Flutter Engine. I discovered they use branches as tags like nixpkgs. I'll check a bit later when I have time.

Reply:
That patch is in 3.7.0 which was released just yesterday. We're using 3.3.7 which seems to be the preferred engine by the Flutter tool but does not contain the patch in one of their stable releases. We'll keep the patch file as I don't want to possibly break anything.

Reply:
If we do change versions, we also have a lot of deps to change in `deps.nix`.

Reply:
yepp, lets just keep this for now. updating should be automated (derive deps.nix from the DEPS file), at least as a future goal. see also https://github.com/input-output-hk/gclient2nix/blob/master/gclient2nix.py and https://chromium.googlesource.com/chromium/tools/depot_tools/+/refs/heads/main/gclient_eval.py

Reply:
Yeah, the goal for this PR is to make it at least build. I think we can get this ready for merging here pretty soon. Last thing I'm working on is some hashes in `deps.nix` for multiple platforms because of how `${platform}` is a variable in DEPS. We also need to fix the linked libraries for the GTK linked library version of Flutter, `libgtk.so` and many other libraries are missing but it was linked.